components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -789,15 +790,28 @@ - (CGSize)intrinsicContentSize {
   }
 }
 
+/**
+ The content size of the tabs in their current layout style.
+
+ For @c FixedJustified:  The content size is the maximum of the bounds within the safe area or the
+                         intrinsic size of the tabs when all tabs have the widest tab's width.
+ For @c FixedClustered*: The bounds within the safe area. This ensures they are positionined
+                         accurately within the content area.
+ For @c Scrollable:      The intrinsic size size of the tabs.
+ */
 - (CGSize)calculatedContentSize {
   switch ([self layoutStyle]) {
     case MDCTabBarViewInternalLayoutStyleFixedJustified: {
-      return [self intrinsicContentSizeForJustifiedLayout];
+      CGSize intrinsicContentSize = [self intrinsicContentSizeForJustifiedLayout];
+      CGSize boundsSize = CGSizeMake(CGRectGetWidth(self.bounds), CGRectGetHeight(self.bounds));
+
+      return CGSizeMake(MAX(boundsSize.width, intrinsicContentSize.width),
+                        MAX(boundsSize.height, intrinsicContentSize.height));
     }
     case MDCTabBarViewInternalLayoutStyleFixedClusteredCentered:
     case MDCTabBarViewInternalLayoutStyleFixedClusteredTrailing:
     case MDCTabBarViewInternalLayoutStyleFixedClusteredLeading: {
-      return [self intrinsicContentSizeForClusteredLayout];
+      return [self availableSizeForSubviewLayout];
```

Comment:
Im not sure if im following this. can you elaborate more on why this is different from the other cases?

also why does intrinsicContentSize also have a similar change?

Reply:
Are you asking why Clustered layouts use the full width of the view rather than the intrinsic content size of that layout? 

I didn't make any changes to `intrinsicContentSize` in this PR. Could you please more detail about what you mean?

Reply:
> Are you asking why Clustered layouts use the full width of the view rather than the intrinsic content size of that layout?
yes

> I didn't make any changes to intrinsicContentSize in this PR. Could you please provide more detail about what you mean?
I saw that intrinsicContentSize and calculatedContentSize do very similar things but now they are drifting apart and im trying to understand why one is changing and not the other.

Reply:
Clustered tabs fill the entire space available (as noted in the summary). This is so that they can be positioned accurately while allowing `contentInset` to behave normally for UIScrollView. This is only really relevant to right-aligned Tabs, which is also noted in the summary.

UIView's `intrinsicContentSize` and UIScrollView's `contentSize` are very different concepts. That they might have been similar was an artifact of the implementation (and partly from this bug). Is there a specific reason why you feel they should not be very different?  I'm sorry, but I don't understand why they might be related.

Reply:
ok sgtm